### DISCUSSION TITLE
Facilitate / Automate hyperlink creation for productivity

### DISCUSSION THREAD

**Discussion by garylemasson** [freeplane/freeplane#745] [Ideas]
Hi, 

I am using freeplane for a lot of things related to work and i miss a productivity feature, i explain :

[see the screenshot (sorry it's in french)
](https://drive.google.com/file/d/1kBeOM5NBorMIu3nclTRTAWrtyaRb7GRI/view?usp=sharing)

**As I organise my ideas, very often i integrate hyperlinks to the web as nodes.**

When I copy paste a URL in a new node, the link is not recognised as a hyperlink automatically.

I finally found the feature (which is manual) : 

- right click on node > insert > insert link > create a link from this node

**4 steps ... to achieve with the mouse only or on the top menu**
And there is no keyboard shortcut assigned to this functionality
(no possibility as well to change the shortcuts and assign it)

**I believe Freeplane miss this functionnality for productivity reasons.**
When you paste a URL inside a node... this is because you want to have access to this url by a click, don't you ?
So why not automating this feature

**Solutions suggested :**
Best solution in my opinion
- Automatically recognize URL links and create a hyperlink to it. All URLs on the Web start with http:// or https// ... It could be also an option to activate from the preference menu...

Or does this feature exist already from a plug in or a feature I have not found yet on the freeplane software ?

Less productive solution
- Add a keyboard shortcut to this feature

**other ideas**
My issue could also appy to email adresses, easy regex to make to recognize them (issue would be to connect to the right application between a Webmail or heavy email client sur as Outlook)
Another possibilty would be to create links to folders on our harddrive and open the finder on mac or the file explorer on windows

I really want to find a productivity solution to this issue...

Can you help me ?

Thanks 

Gary

**Comment by EdoFro** [comment]
which Freeplane version are you using?

In my computer it works as you expect.

1. I copy the link (as text from chrome)
![image](https://user-images.githubusercontent.com/35700575/195873131-3fe3a3c4-a8e9-4164-9979-bc3908d74dff.png)

2. I go to freeplane, select an existing node ...
![image](https://user-images.githubusercontent.com/35700575/195873480-d61109a6-0c9f-47e2-bf44-f0bb59ce3987.png)

3. ... and hit Ctrl+V to paste it andit creates a new node with the url in its text and as a link
![image](https://user-images.githubusercontent.com/35700575/195873897-057161fa-1aef-445f-8a59-181ae02f2b4b.png)

It beheaves the way you dislike if you create the node first and paste the URL as text of the new node
![image](https://user-images.githubusercontent.com/35700575/195874255-29286295-63f9-431a-a6a2-af7181b110c1.png)

![image](https://user-images.githubusercontent.com/35700575/195874347-60095454-0838-45f2-9551-43f228872eda.png)

![image](https://user-images.githubusercontent.com/35700575/195874409-4b6cec8d-268b-430a-9904-52f3e749cee3.png)


You can also add a hyperlink by clicking on it in chrome and drag and drop it into freeplane

 
![image](https://user-images.githubusercontent.com/35700575/195875748-201e5d5e-09c9-48d5-b069-221b984b2347.png)


![image](https://user-images.githubusercontent.com/35700575/195875438-7bbe16f2-e993-4162-a155-b0547fa665f1.png)

**Reply by garylemasson** [reply]
ahhh ok thanks...
I was actually opening a new node and pasting it inside the text area... which did not work
drag and drop works... that solves my problem
and pasting on parent node too

really thanks for the tip ^^

Best regards

**Comment by EdoFro** [comment]
Maybe a possible improvement when pasting a link into Freeplane (for example in case Drag and Drop) would be that the new node uses the url as link but the link's description as node's new text.

For example, in this case:

---

![image](https://user-images.githubusercontent.com/35700575/195877738-3e2a6283-36ac-4e59-a6ee-245e72226a11.png)

---

![image](https://user-images.githubusercontent.com/35700575/195877982-5ba40aff-fc67-43d4-94e7-f477c1242c87.png)

---

currently:
![image](https://user-images.githubusercontent.com/35700575/195878107-f760e2b2-47fa-4ae0-8f33-04b65027aead.png)


---
¿maybe better?
![image](https://user-images.githubusercontent.com/35700575/195878709-9a51266e-1b3c-4c26-a004-ac16783e2d92.png)

**Reply by garylemasson** [reply]
Hi thanks for the tip :)
